### PR TITLE
Fix: build JS files first during npm start

### PR DESCRIPTION
**Pull request by marissahuysentruyt**
This PR corrects some build processes regarding JS in this project:

- removes unnecessary passthrough functions in `eleventy.js`
- updates scripts for JS
- updates `base.njk` script `src`
- creates` index.js` to import all scripts in one place

### Validation
- [ ] Pull down the branch
- [ ] Run `npm install` then `npm start`
- [ ] Visit the [homepage](http://localhost:8080/)
- [ ] Verify the `animate-text.js` file is running in the hero section as expected
- [ ] In your code editor, add a `console.log` to one of the `.js` files. Revisit the browser console and verify the build has included your change
- [ ] In your code editor, make an SCSS change that will affect the homepage. Revisit the browser and verify the build has included your change

### CODE REVIEW

**Comment by marissahuysentruyt** [comment]
I would love some help writing validation steps for this PR. Would be something along the lines of changing things in the `package.json` scripts and making sure the output is in the `dist` folder?

File: package.json
Comment (dustin-jw) [inline package.json:1]: This isn't related to anything in this PR, but I noticed that both `lint:js` and `lint:scss` fail. One of the first things I do when I set up linting scripts in a [GitHub workflow](https://github.com/dustin-jw/eleventy-starter/blob/main/.github/workflows/lint-and-test.yml) to make sure that I don't introduce any linting errors in future PRs.

**Review by dustin-jw — COMMENTED** [review]
I just have a couple of minor suggestions, but in terms of validation instructions, I'd probably write something like this:

1. Pull down this branch
2. Run `npm start`
3. Visit the home page, confirming that the animation works as expected
4. Add a `console.log` statement or otherwise edit a JS file, confirming that your changes are built and reflected in the site when you save the file
5. Change a property in a SCSS file to something obvious, like `background-color: hotpink`, confirming that the changes are built and take effect when you save the file

File: package.json
Comment (marissahuysentruyt) [reply package.json:1]: Yes! That's actually something I'm looking into right now. I think we did some github workflow stuff during the capstone project, and I've been talking with Jon about learning more about github actions in general, so...it's on my to-do list to look into.

File: src/layouts/base.njk [hunk 1]
```diff
@@ -162,6 +162,6 @@
     <p>&copy;&nbsp;2023 Marissa Huysentruyt</p>
   </footer>
 
-  <script src="scripts/animate-text.js" defer type="module"></script>
+  <script src="scripts/index.js" type="module"></script>
```
[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply src/layouts/base.njk]: Ah, thank you. Should be fixed here: https://github.com/marissahuysentruyt/marissahuysentruyt/pull/20/commits/8417e6020456a1758b22d7f75033d7142949b1fe

File: package.json [hunk 2]
```diff
@@ -5,12 +5,12 @@
...
     "build:styles": "sass src/scss/main.scss:dist/css/main.css --no-source-map",
-    "watch:scripts": "esbuild src/scripts/*.js --format=esm --bundle=true --splitting=true --outdir=dist --watch --sourcemap=inline",
-    "build:scripts": "esbuild src/scripts/*.js --format=esm --bundle=true --splitting=true --outdir=dist --minify=true",
+    "watch:scripts": "esbuild src/scripts/*.js --format=esm --bundle=true --outdir=dist/scripts --watch --sourcemap=inline",
+    "build:scripts": "esbuild src/scripts/index.js --format=esm --bundle=true --outfile=dist/scripts/index.js --minify=true",
     "watch:eleventy": "npx @11ty/eleventy --serve --incremental",
     "build:eleventy": "npx @11ty/eleventy",
     "prestart": "rm -rf ./dist",
-    "start": "npm-run-all prestart --parallel build:styles --parallel watch:*",
+    "start": "npm-run-all prestart --parallel build:styles --parallel build:scripts --parallel watch:*",
```
[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply package.json:13]: So I could delete the `prestart` script entirely, and then my `start` script is just `run-p build:* watch*`. cool!

[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply package.json]: Updated: https://github.com/marissahuysentruyt/marissahuysentruyt/pull/20/commits/52d9dfe68e672e4e0b63adae8b63802455d7cd1c

**Review by dustin-jw — APPROVED** [review]
Looks good! Ship it 🚢

**Pull request merged**